File: dotnet/GHIElectronics.DUELink/Touch.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GHIElectronics.DUELink {
    public partial class DUELinkController {


        public class TouchController {

            SerialInterface serialPort;

            public TouchController(SerialInterface serialPort) => this.serialPort = serialPort;

            public bool Touch(int pin, int charge_t, int charge_s, int timeout) {
                var cmd = $"touch({pin}, {charge_t}, {charge_s}, {timeout})";
                this.serialPort.WriteCommand(cmd);

                var res = this.serialPort.ReadResponse();
                var val = false;
                if (res.success) {


                    try {
                        val = int.Parse(res.response) == 1 ? true : false;
                    }
                    catch { }
                }

                return val;

            }

           
            
        }
    }
}

File: micropython/test/duelink/touch.py
class TouchController:
    def __init__(self, serialPort):
        self.serialPort = serialPort

    def Touch(self, pin: int, charge_t: int, charge_s: int, timeout: int):
        cmd = "touch({0}, {1}, {2}, {3})".format(pin, charge_t, charge_s, timeout)
        self.serialPort.WriteCommand(cmd)

        r,s = self.serialPort.ReadResponse()
        
        val = False
        if r:
            try:
                return s
            except:
                pass
        return val
